simulation/traffic_simulator/src/entity/ego_entity.cpp

```diff
@@ -207,15 +213,16 @@ auto EgoEntity::getEntityStatus(const double time, const double step_time) const
       return accel;
     }();
 
-    const auto route_lanelets = getRouteLanelets();
+    auto route_lanelets = getRouteLanelets();
+    const auto unique_route_lanelets = traffic_simulator::helper::getUniqueValues(route_lanelets);
```

Comment:
route lanelets is passed to getUniqueValues function once.
So, I suggest this  change.

```suggestion
    const auto unique_route_lanelets =
      traffic_simulator::helper::getUniqueValues(getRouteLanelets());
```